compiler/rustc_codegen_llvm/src/llvm/mod.rs

```diff
@@ -331,28 +332,43 @@ pub fn last_error() -> Option<String> {
     }
 }
 
-pub struct OperandBundleDef<'a> {
-    pub raw: &'a mut ffi::OperandBundleDef<'a>,
+/// Owns an [`OperandBundle`], and will dispose of it when dropped.
+pub(crate) struct OperandBundleOwned<'a> {
+    raw: ptr::NonNull<OperandBundle<'a>>,
```

Comment:
ah, and the `&'a Value` is what means we can't just make `OperandBundleOwned<'a>` be actually this, right?
```rust
type OperandBundleOwned = OperandBundle<'static>;
```

Reply:
Yeah, from what I can tell an operand bundle holds copies of the value pointers internally, so it would be unsound for the bundle to outlive those values.

(In practice, I think the lifetime of values is always `'ll`, i.e. the lifetime of the LLVM context.)

Reply:
Oh one other relevant point, `OperandBundleOwned` is closer to `Box<OperandBundle>`, since its representation is a pointer to the underlying bundle, and the owner is responsible for deallocating on drop.

So the two types would still have to be different, unless we hypothetically had `&own` references to handle dropping.